action/protocol/execution/evm/evm.go

```diff
@@ -627,6 +627,7 @@ func SimulateExecution(
 		protocol.ActionCtx{
 			Caller:     caller,
 			ActionHash: hash.Hash256b(byteutil.Must(proto.Marshal(ex.Proto()))),
+			ReadOnly:   true,
```

Comment:
also set this flag in `ReadContract`? then PR4492 can take advantage of this and no need to change `prepareStateDB`

Reply:
ReadContract also calls it